Content.Server/Radio/EntitySystems/JammerSystem.cs

```diff
@@ -102,12 +139,64 @@ private bool ShouldCancelSend(EntityUid sourceUid)
 
         while (query.MoveNext(out _, out _, out var jam, out var transform))
         {
-            if (source.InRange(EntityManager, _transform, transform.Coordinates, jam.Range))
+            if (source.InRange(EntityManager, _transform, transform.Coordinates, GetCurrentRange(jam)))
             {
                 return true;
             }
         }
 
         return false;
     }
+
+    private void OnGetVerb(Entity<RadioJammerComponent> entity, ref GetVerbsEvent<Verb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract)
+            return;
+
+        var user = args.User;
+
+        byte index = 0;
+        foreach (var setting in entity.Comp.Settings)
+        {
+            // This is because Act wont work with index.
+            // Needs it to be saved in the loop.
+            var currIndex = index;
+            var verb = new Verb
+            {
+                Priority = currIndex,
+                Category = VerbCategory.PowerLevel,
+                Disabled = entity.Comp.SelectedPowerLevel == currIndex,
+                Act = () =>
+                {
+                    entity.Comp.SelectedPowerLevel = currIndex;
+                    var jammerComp = Comp<DeviceNetworkJammerComponent>(entity.Owner);
+                    jammerComp.Range = GetCurrentRange(entity.Comp);
+                    jammerComp.Dirty();
+                    _popup.PopupEntity(Loc.GetString(setting.Message), user, user);
+                },
+                Text = Loc.GetString(setting.Name),
+            };
+            args.Verbs.Add(verb);
+            index++;
+        }
+    }
+    private static float GetCurrentWattage(RadioJammerComponent jammer)
+    {
+        return jammer.Settings[jammer.SelectedPowerLevel].Wattage;
+    }
+    private static float GetCurrentRange(RadioJammerComponent jammer)
+    {
+        return jammer.Settings[jammer.SelectedPowerLevel].Range;
+    }
+    private void ChangeLEDState(bool isLEDOn, EntityUid uid, RadioJammerComponent? component = null,
+        AppearanceComponent? appearance = null)
+    {
+        _appearance.SetData(uid, RadioJammerVisuals.LEDOn, isLEDOn, appearance);
+    }
```

Comment:
You don't seem to need the RadioJammerComponent here. If you want to verify that the entity does indeed have the component then you should do something like this:

if (!Resolve(uid, ref component)
    return;

Reply:
You are completely correct! I think I was actually using the component in the past but at some point I stopped using it. I don't really know if I should bother checking it or if I should just remove the entire thing input. What do you think I should do?

Reply:
I mean since you are calling it only inside the system, and the function wouldn't have any horrible effects on things without the component you should probably remove  the check. Every place where its called from it's basically already guaranteed the component would exist anyways.